### DISCUSSION TITLE
loop in automations

### DISCUSSION THREAD

**Discussion by wickermann165** [Budibase/budibase#5464] [Help]
Hi,
I have following use-case in mind but do not know how to proceed in budibase.

We have a CRM system with REST API and I would like to take a subset of the data from CRM and make a local copy of the data after some transformations in budibaseDB for use in an app. The goal is that the app synchronizes once a week with the CRM (might be done even manually by clicking on a button, it does not matter).

I am able to get the data from the CRM with REST API data connector and query. However I am not able to store the data in the local budibseDB. It seems that I would need to be able to loop in the automation as the result of the REST API query is an array of table rows from the CRM database and I need to add a row in the budibaseDB for each of the rows returned by the query. It seems that the budibase automation does not support looping (there is no node for it in the automation). I should be able to loop in a JS scripting node but I have not been able to manipulate the database from the JS scripting node (from forum search it seems that it is on purpose as the JS is running in some kind of sandbox).

From #4416 It seems that loop is not an option, but is planned to be implemented later #4427. 

Does anybody have an idea how to move forward with my idea without loop in automation?

Thanks for any help!

**Comment by melohagan** [comment]
Hi @wickermann165 

One possibility for the time being could be to store the array of data into a JSON type column: https://docs.budibase.com/docs/json

Otherwise it appears the Automation For Loop is well underway: https://github.com/Budibase/budibase/pull/5145
